### PR TITLE
Better error message when custom objective specified as string without required args in pipeline score

### LINKED ISSUE #1603: Improve error msg when non-string objective provided without required args
From December Blitz: https://alteryx.quip.com/hifwAO0YHqul/Blitz-December-2020-Notes

image (6)

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description
Fixes #1603 

```python
pipeline.score(X, y, objectives=['cost benefit matrix'])
```

image

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: evalml/tests/pipeline_tests/test_pipelines.py [hunk 1]
```diff
@@ -1985,3 +1992,28 @@ def test_pipeline_class(pipeline_class):
...
+    pipeline.fit(X, y)
+    msg = "Cannot pass cost benefit matrix as a string in pipeline.score. Instantiate first and then add it to the list of objectives."
+    with pytest.raises(ObjectiveCreationError, match=msg):
+        pipeline.score(X, y, objectives=["cost benefit matrix", "F1"])
+
+    # Verify ObjectiveCreationError only raised when string matches an existing objective
+    with pytest.raises(ObjectiveNotFoundError, match="cost benefit is not a valid Objective!"):
+        pipeline.score(X, y, objectives=["cost benefit", "F1"])
+
+    # Verify no exception when objective properly specified
```
Comment (angela97lin) [inline evalml/tests/pipeline_tests/test_pipelines.py:2017]: Curious, what's the behavior if CostBenefitMatrix is not properly instantiated, like `CostBenefitMatrix(1)`? :o

**Review by angela97lin — APPROVED** [review]
LGTM! 🥳  🎉 

Left a comment more for my own understanding heh

File: evalml/tests/pipeline_tests/test_pipelines.py (hunk 1 above)
Comment (freddyaboulton) [reply evalml/tests/pipeline_tests/test_pipelines.py:2017]: You'd get a `TypeError` but before the code gets into the `score` method!

**Review by chukarsten — APPROVED** [review]
Solid, good work.

**Pull request merged**